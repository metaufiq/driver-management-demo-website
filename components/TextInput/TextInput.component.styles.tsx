import styled from "styled-components";

export const StyledInput = styled.input`
  border: 0;
  :focus{
    outline: none;
  }
`

export const Container = styled.div`
  font-size: 1rem;
  padding: 1rem;
  border: 1px solid #f2f2f2;
  border-radius: 0.125rem;
  flex-direction: row;
  display: flex;
`

export const LeftIconContainer = styled.div`
  font-size: 1rem;
  padding-top: 0.125rem;
  padding-right: 0.5rem;
  color: #ff453b;
`