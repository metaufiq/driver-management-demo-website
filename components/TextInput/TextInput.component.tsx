import { StyledInput } from "./TextInput.component.styles";
import { Props } from "./TextInput.component.types";

const TextInput = (props: Props)=>{
  return(
    <StyledInput
      id={props.id}
      type={props.type}
      placeholder={props.placeholder}
      onChange={props.onChange}
    />
  )
}

export default TextInput;