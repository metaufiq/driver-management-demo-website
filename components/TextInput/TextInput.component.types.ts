import { ChangeEventHandler, HTMLInputTypeAttribute } from "react"

export type Props = {
  placeholder: string,
  id?: string,
  type: HTMLInputTypeAttribute,
  leftIcon?: JSX.Element,
  onChange: ChangeEventHandler<HTMLInputElement>
}