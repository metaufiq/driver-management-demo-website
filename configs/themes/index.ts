const themes = {
  colors: {
    primary: '#ff453b'
  },
};

export { themes };