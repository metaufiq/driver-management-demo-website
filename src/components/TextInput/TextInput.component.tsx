import { ReactElement } from "react";

import { Component } from "../../../index.types";
import { Container, LeftIconContainer, StyledInput } from "./TextInput.component.styles";
import { Props } from "./TextInput.component.types";

/**
 * Text Input components
 * @param {Props} props component props 
 * @returns {ReactElement} Text Input components
 */
const TextInput:Component<Props> = (props: Props): ReactElement=>{
  return(
    <Container>
      {props.leftIcon && <LeftIconContainer>
        {props.leftIcon}
      </LeftIconContainer>}
      
      <StyledInput
        type={props.type}
        placeholder={props.placeholder}
        onChange={props.onChange}
      />
    </Container>
  )
}

export default TextInput;