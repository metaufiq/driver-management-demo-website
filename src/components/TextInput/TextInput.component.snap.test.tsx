import { themes } from "../../configs/themes";
import { assertSnapshots } from "../../test";
import { Configs } from "../../test/index.types";
import TextInput from ".";
import { Props } from "./TextInput.component.types";

describe('TextInputSnapshot', () => {
  const mockProps:Props = {
    onChange:()=>{},
    placeholder: 'placeholder',
    type: 'password'
  };

  const configs:Configs<Props> = [
    {
      props: mockProps,
      desc: 'should render normal TextInput',
      useTheme: true,
      theme: themes
    },
    {
      props: { ...mockProps, leftIcon: <div/>},
      desc: 'should render normal TextInput with left icon',
      useTheme: true,
      theme: themes
    }
  ];

  assertSnapshots(TextInput, configs);
});
