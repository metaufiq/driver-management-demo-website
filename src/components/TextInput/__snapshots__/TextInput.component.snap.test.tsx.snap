// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputSnapshot should render normal TextInput 1`] = `
<div
  className="sc-gsnTZi hEqjwc"
>
  <input
    className="sc-bczRLJ gBziFm"
    onChange={[Function]}
    placeholder="placeholder"
    type="password"
  />
</div>
`;

exports[`TextInputSnapshot should render normal TextInput with left icon 1`] = `
<div
  className="sc-gsnTZi hEqjwc"
>
  <div
    className="sc-dkzDqf gZKilg"
  >
    <div />
  </div>
  <input
    className="sc-bczRLJ gBziFm"
    onChange={[Function]}
    placeholder="placeholder"
    type="password"
  />
</div>
`;
